### PR TITLE
ibc-go v5: cosmos-sdk 46

### PR DESCRIPTION
**Pull request by faddat**
## Description

Cosmos sdk v46 changes to ibc-go

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by nghuyenthevinh2000** [comment]
can anyone approve this workflow please

**Comment by faddat** [comment]
* https://github.com/strangelove-ventures/ibctest/pull/209

**Comment by crodriguezvega** [comment]
@faddat can we get this branch updated with the latest from main and have any conflicts solved?

**Comment by faddat** [comment]
@crodriguezvega kinda blocked upstream right now.  Issue is in ibctest.  Will see what I can do about it.

**Comment by faddat** [comment]
I'm going to give docker-build a condiditional so it doesn't try to login to the docker registry when building from a branch.

**Comment by faddat** [comment]
This branch has been gofumpted and goimports'd and fails.  

The command `make goimport` results in:

`Error, path is required but not specified`

**Comment by faddat** [comment]
https://github.com/bianjieai/ibc-go/pull/1

**Comment by crodriguezvega** [comment]
> @crodriguezvega kinda blocked upstream right now. Issue is in ibctest. Will see what I can do about it.
> 
> * [ibctest blocks current 46 upgrade path #1706](https://github.com/cosmos/ibc-go/issues/1706)
> 
> Maybe another solution is to use a second go.mod for e2e? Not sure the right path.

Hey @faddat. The e2e tests use a different go.mod now (see [this PR](https://github.com/cosmos/ibc-go/pull/1701)). It was merged very recently, so maybe you didn't fetch the latest from our main? Hopefully this helps and unblocks for way...

**Comment by crodriguezvega** [comment]
> I'm going to give docker-build a condiditional so it doesn't try to login to the docker registry when building from a branch.

Thanks, this is a good point!

**Comment by crodriguezvega** [comment]
> This branch has been gofumpted and goimports'd and fails.
> 
> The command `make goimport` results in:
> 
> `Error, path is required but not specified`

So if I understand correctly, it would wise to merge first [this PR](https://github.com/cosmos/ibc-go/pull/1717) with all the fumpt changes first, right?

Regarding the goimports step, can we also add this in a separate PR? We would really appreciate if this PR contains only the strictly necessary changes to upgrade to 0.46 and nothing else. That will help us to review faster. I hope that's ok... Thanks in advance!

**Comment by chatton** [comment]
> I'm going to give docker-build a condiditional so it doesn't try to login to the docker registry when building from a branch.

Hi @faddat, sorry for not chiming in sooner. We actually **do** want ot build and push docker images on branches. This is so we can run e2e-tests with the code changes on branches before merging to main. However as the e2e tests are still a WIP I think we can skip the docker build phase, but not for all branches.

The reason it's failing is because github action secrets aren't passed to workflows triggered by forks by default. We can update the docker build job to depend on a secret e.g. `secrets.GITHUB_TOKEN`

We should be able to do this with something like

```yaml
   env:
     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   if: "${{ env.GITHUB_TOKEN != '' }}"
```

As far as I'm aware it is not possible to check for secrets directly in the `if` (but I am happy to be wrong here!)

File: testing/simapp/simd/cmd/genaccounts.go
Comment (crodriguezvega) [inline testing/simapp/simd/cmd/genaccounts.go:68]:
```suggestion
					return fmt.Errorf("failed to get address of the record: %w", err)
```

File: testing/simapp/simd/cmd/root.go
Comment (crodriguezvega) [inline testing/simapp/simd/cmd/root.go:10]: Can you please remove the empty line here? We like to group SDK/Tendermint imports together.

File: testing/simapp/app.go
Comment (crodriguezvega) [inline testing/simapp/app.go:139]: nit: can we maybe break this long line in one line for each element of the slice?

**Review by crodriguezvega — COMMENTED** [review]
I think it's looking good! Thanks, @faddat!

I have left a few comments and a couple of questions. Besides those, two more things:

- I think the file `testing/sdk_test.go` can be completely removed (see comments [here](https://github.com/cosmos/ibc-go/pull/1205#discussion_r838552243)).
- Do you mind to copy [this implementation](https://github.com/cosmos/ibc-go/blob/carlos/upgrade-sdk-0.46-tendermint-0.35/testing/simapp/ante_handler.go#L20-L48) of the `NewAnteHandler` function in `testing/simapp/ante_handler.go` please?

File: testing/simapp/simd/cmd/root.go
Comment (faddat) [reply testing/simapp/simd/cmd/root.go:10]: yep

File: modules/apps/transfer/simulation/genesis_test.go
Comment (faddat) [reply modules/apps/transfer/simulation/genesis_test.go:24]: @nghuyenthevinh2000 / @vuong177 I forget who fixed it, I think maybe vinh.  

Sure did make it work, though.

File: .gitpod.yml
Comment (faddat) [reply .gitpod.yml:1]: you're right, it's just an artifact.

Thank you!

File: .github/workflows/test.yml
Comment (faddat) [reply .github/workflows/test.yml:106]: I.... have no idea.

Will check out main's.

File: modules/apps/29-fee/ibc_middleware.go
Comment (faddat) [reply modules/apps/29-fee/ibc_middleware.go:251]: hum.  I wonder what I've done to my vscode.  I save this file, and it immediately flips over to the changed version.

File: testing/simapp/app.go
Comment (faddat) [reply testing/simapp/app.go:773]: fact!  Thanks!

File: testing/simapp/app.go
Comment (faddat) [reply testing/simapp/app.go:139]: absolutely, I prefer to do things that way too.

File: testing/simapp/app.go
Comment (faddat) [reply testing/simapp/app.go:314]: this is why we'd done it the other way originally. 

thoughts?

File: modules/apps/29-fee/ibc_middleware.go
Comment (faddat) [reply modules/apps/29-fee/ibc_middleware.go]: neovim to the rescue....

**Comment by faddat** [comment]
@crodriguezvega 

✅ I think the file testing/sdk_test.go can be completely removed (see comments https://github.com/cosmos/ibc-go/pull/1205#discussion_r838552243).
✅ Do you mind to copy [this implementation](https://github.com/cosmos/ibc-go/blob/carlos/upgrade-sdk-0.46-tendermint-0.35/testing/simapp/ante_handler.go#L20-L48) of the NewAnteHandler function in testing/simapp/ante_handler.go please?

**Comment by chatton** [comment]
@faddat  it looks like there's a bit of a chicken and egg problem with the e2e go.mod since ibc-go v5 does not exist yet. For now can we leave  the e2e go.mod at `v4`, we we can update it in a follow up?

**Comment by faddat** [comment]
Thanks @vuong177!

I think that this is now ready to go. 

Ofc, it isn't stable yet.

@charleenfei @crodriguezvega -- the next question to me, surrounds a few other PR's:

* #1418 - SetUpgradedConsensus state is a question here.  I don't know how the main branch does not anger codeql when doing SetUpgradedConsensusState for example. My dev env treats the linter rules on IBC-go as kings, so I must admit that I've already done some of the linting work in this branch and thus, getting #1709 in is now quite important to me. 

* #1709  -  question here is about Robert's thinking on a 46-compatible ibc v3.x trouble is that I want & like standards and don't want the relay network exploding.... so, I want to get the ibc-go maintainer's thoughts on this -- as well as Robert's.  

* #1510 - community and our clients definitely want NFTs. This work by the iris team is planned for use in:
  * Iris
  * Pylons
  * Dig
  * Craft Economy
  * Stargaze
  * Gravity Bridge
  * Juno

**Comment by crodriguezvega** [comment]
Hey @faddat. 

- I think it would be good to merge #1418 as well soon so that we can release its API breaking changes as part of v5. I will do another round of review today.
- I just posted a comment on #1709. I think it is best to close the PR and delete the `release/v3.3.x` branch.
- Regarding #1510. We have had several discussions in the last few weeks about adding new apps to ibc-go, and we came to the conclusion that it is best for the ecosystem if the apps, middleware or light clients developed by other teams reside in a repo of their own organizations instead of in ibc-go. I have messaged Zhiqiang on Slack about this, but didn't hear from him yet... The reasoning behind this decision is that we want to prevent exactly what it is happening now: we are busy with lots of other things and we're not having enough time to properly review on time the NFT app and that's blocking its release and its usage by the rest of the chains in the ecosystem. We will write a full disclosure of our vision for ibc-go and add it to the repo, so that the rest of the ecosystem is aware of this.

**Comment by chatton** [comment]
@faddat it looks like the e2e is still failing for a new reason, I don't to this to block this PR, there are obviously a few issues that we need to iron out to make this work correctly on forks. I will try and fix these issues in a separate PR once this is merged!

**Comment by faddat** [comment]
@crodriguezvega OK, I can close #1709 

Regarding the NFTs, I like the registry idea, but I think that can lead to this:

* https://github.com/cosmos/cosmos-sdk/issues/12228

As a user, that's my number one issue with the sdk and IBC.  I think I would like to suggest that it's a good idea to have the NFTs here-- as well as other IBC apps. 

I know "modular" is "the way" but erm, libp2p.

I don't know if you have ever tried to use it, personally I love it, except for the way that it is modularized.  Since we (notional) and others really do want a 46 compatible ibc-go, We could drop the NFTs, but I'd encourage you to consider maybe outsourcing more review, to the ci system, perhaps with the strangelove e2e tests. 

Also may make sense to formalize the community role in maintaining different modules. 

* https://github.com/cosmos/cosmos-sdk/issues/12228

Is the sole reason I'm kinda sad that ibc-go decoupled from the sdk.  It made integration much harder, and in the end, we ended up just doing it, as the teams were otherwise engaged.  

NB: not a blame fest here that you guys are busy-- I get it and very much. 

More looking for solutions.  Maybe the registry you're describing is that -- but maybe not.  I am concerned about compatibility.

**Comment by faddat** [comment]
>  what do you think?

I think the reasoning is very sound. 

Let's do that

**Comment by crodriguezvega** [comment]
@faddat I think we need to fix the imports in `/ibc-go/modules/apps/transfer/types/trace.go` so that the `Goimports` step in CI doesn't fail. Would you manage to do that?

**Comment by faddat** [comment]
FYI, make goimports didn't catch that.

Seems fumpt did though? 

Let's see if CI likes us now...

**Comment by faddat** [comment]
> i agree that it takes a bit more work to make sure things aren't breaking and appreciate you/your teams efforts to support us in the integration PRs, but my personal opinion is that i am not entirely convinced that doing the work of merging ibc-go back into the sdk and trying for a monorepo will necessarily be a time saver at this point (though i could be wrong, i cannot speak for everyone that would be involved in this effort).


I don't think that we can/should return to a monorepo.  Hoping to automate some of the pain away, though.

Also, as a thought: so I have this sdk issue:

* https://github.com/cosmos/cosmos-sdk/issues/12228

That's where I'd like to see this headed I think.

**Comment by faddat** [comment]
I think that this is complete now, too.

**Comment by faddat** [comment]
https://app.circleci.com/pipelines/github/CosmWasm/wasmd/2215/workflows/6ad06175-e8dc-4f12-804f-2a02f8882018/jobs/7537

https://github.com/CosmWasm/wasmd/pull/912

One note on this-- while upgrading cosmwasm to v46, I discovered that we've diverged from the sdk -- if you look through there at `helpers.gentx` -- in the sdk we've moved over to `helpers.gensignedmocktx` and that includes a time variable.  Curious if you've any input on this.

**Comment by crodriguezvega** [comment]
@faddat Thanks for bumping also the go package version to v5. However, would you mind creating a new PR that isolates the changes for the bump to v5? This PR with the upgrade to SDK 0.46 contains now changes in more than 400 files and the review is becoming very difficult. 

We have created this [issue](https://github.com/cosmos/ibc-go/issues/1780) for bumping to v5. We can open the PR ourselves, if you prefer. If we merge the PR with the bump to v5 first, then you don't need to revert the bump to v5 here (git should be clever enough).

**Comment by robert-zaremba** [comment]
Can we have any estimate of having 0.46 compatible release? Would be great if v5 will focus only on that.

**Comment by faddat** [comment]
@crodriguezvega great point, and yes, I'm happy to do that.

File: testing/simapp/app.go
Comment (tomtau) [reply testing/simapp/app.go:314]: `GetSealedConfig(...).GetBech32AccountAddrPrefix()` ?

**Comment by ilgooz** [comment]
Hey, Ignite CLI team is also waiting on this upgrade to be able to ship a new release to support v0.46.0. 🎈

**Comment by faddat** [comment]
That's huge!  I'd love to have a call sometime, I have some thoughts on the design of the template (templates, in fact)

**Comment by faddat** [comment]
@ilgooz -- this could be useful to you, too.

It does need some touching up:

https://github.com/CosmWasm/wasmd/pull/912

**Comment by faddat** [comment]
* https://github.com/cosmos/ibc-go/pull/1653/commits/0a11156cba034c5d424b59b7eaf367a8e3ce0b29

Is why we should add gitpod, btw.

I've got an image where stuff works, configured quite like Vuong's computer in fact....

@charleenfei About .pb.go -- if there were import ordering changes, they're accidental.  At one point here, I flubbed using goimports.  Gofumpt replaces goimports and we've no longer got a need to use / think about goimports.  Gofumpt will also keep away from .pb.go

**Comment by faddat** [comment]
```
[gitpod@notionallabs-ibcgo-uki9c679s85 ibc-go]$ gofumpt -w -l .
e2e/fee_middleware_test.go
[gitpod@notionallabs-ibcgo-uki9c679s85 ibc-go]$ make proto-update-deps
[gitpod@notionallabs-ibcgo-uki9c679s85 ibc-go]$ make proto-all
Formatting Protobuf files
Unable to find image 'tendermintdev/docker-build-proto:latest' locally
latest: Pulling from tendermintdev/docker-build-proto
97518928ae5f: Pull complete 
47ceeaf0160f: Pull complete 
f2f60ec567da: Pull complete 
cceaf9bbff94: Pull complete 
Digest: sha256:b2bfae1f1fbdaece7c14a692e522d5bfb92024a1fb5a610fc059dd9ad2f18f4f
Status: Downloaded newer image for tendermintdev/docker-build-proto:latest
Unable to find image 'bufbuild/buf:1.0.0-rc8' locally
1.0.0-rc8: Pulling from bufbuild/buf
a0d0a0d46f8b: Pull complete 
962721a90331: Pull complete 
c86c0bb602ab: Pull complete 
Digest: sha256:d3674a94d2d7cab6cc1e881e2636576bc1de3f0a510746e838dc63e73c5bc525
Status: Downloaded newer image for bufbuild/buf:1.0.0-rc8
{"path":"third_party/proto/confio/proofs.proto","start_line":1,"start_column":1,"end_line":1,"end_column":1,"type":"COMPILE","message":"syntax error: unexpected int literal"}
{"path":"third_party/proto/cosmos/base/v1beta1/coin.proto","start_line":20,"start_column":8,"end_line":20,"end_column":8,"type":"COMPILE","message":"field cosmos.base.v1beta1.Coin.amount: unknown extension cosmos_proto.scalar"}
{"path":"third_party/proto/cosmos/base/v1beta1/coin.proto","start_line":32,"start_column":8,"end_line":32,"end_column":8,"type":"COMPILE","message":"field cosmos.base.v1beta1.DecCoin.amount: unknown extension cosmos_proto.scalar"}
{"path":"third_party/proto/cosmos/base/v1beta1/coin.proto","start_line":37,"start_column":19,"end_line":37,"end_column":19,"type":"COMPILE","message":"field cosmos.base.v1beta1.IntProto.int: unknown extension cosmos_proto.scalar"}
{"path":"third_party/proto/cosmos/base/v1beta1/coin.proto","start_line":42,"start_column":19,"end_line":42,"end_column":19,"type":"COMPILE","message":"field cosmos.base.v1beta1.DecProto.dec: unknown extension cosmos_proto.scalar"}
make: *** [Makefile:405: proto-lint] Error 100
[gitpod@notionallabs-ibcgo-uki9c679s85 ibc-go]$ make proto-gen
Generating Protobuf files
Unable to find image 'tendermintdev/sdk-proto-gen:v0.2' locally
v0.2: Pulling from tendermintdev/sdk-proto-gen
540db60ca938: Pull complete 
adcc1eea9eea: Pull complete 
4c4ab2625f07: Pull complete 
0510c868ecb4: Pull complete 
afea3b2eda06: Pull complete 
20ff8f5296f2: Pull complete 
fa3aca281e67: Pull complete 
0881ba42b10c: Pull complete 
2a9ac9ddce2c: Pull complete 
d3511612d302: Pull complete 
Digest: sha256:4d0bca1de3a6169446d418c955fafcd3028c9b97a874dcd45271d8bd74f80e9a
Status: Downloaded newer image for tendermintdev/sdk-proto-gen:v0.2
third_party/proto/cosmos/base/v1beta1/coin.proto:20:8:field cosmos.base.v1beta1.Coin.amount: unknown extension cosmos_proto.scalar
third_party/proto/cosmos/base/v1beta1/coin.proto:32:8:field cosmos.base.v1beta1.DecCoin.amount: unknown extension cosmos_proto.scalar
third_party/proto/cosmos/base/v1beta1/coin.proto:37:19:field cosmos.base.v1beta1.IntProto.int: unknown extension cosmos_proto.scalar
third_party/proto/cosmos/base/v1beta1/coin.proto:42:19:field cosmos.base.v1beta1.DecProto.dec: unknown extension cosmos_proto.scalar
make: *** [Makefile:391: proto-gen] Error 100
[gitpod@notionallabs-ibcgo-uki9c679s85 ibc-go]$ 
```

so, that's the real proto issue......

**Comment by faddat** [comment]
So basically in gitpod I have the exact ideal environment to work on this kind of stuff, and it goes a lot faster.

**Comment by faddat** [comment]
Time for me to ✈️  to seoul.

Will refresh things from the airport if I get the chance

**Comment by colin-axner** [comment]
Hi all, 

The general plan is to:
- merge #1418 
- merge this pr #1653 
- create a release branch and start the release process for v5

We will create a release tracking issue in similar spirit to ones made for v4 (#396 and #1757) in the coming days. This issue will walk through all the steps we will be taking to finalize v5. Thanks for your patience

**Comment by faddat** [comment]
Hey Colin, I'm at the airport now-- could you let me know what's missing from this?

* document the api breaking changes shared between v4 and v5 (eg the ones caused by the linter)
* Migration docs v3 -> v4
* Migration docs v3 -> v5
* Migration docs v4 -> v5

For 4 to 5 and 3 to 5, take the sdk's 44 -> 46 and 45 -> 46 docs and mix them in?

I think I might be able to be a pretty singificant help with that if I spend a few hours on it tomorrow or the day after.

In the meantime, I will keep these PR's warm and thanks to the ibc-go team for the deep tour of the code and practices.

It was fantastic.

**Comment by faddat** [comment]
FYI, this failure: https://github.com/cosmos/ibc-go/runs/7556068869?check_suite_focus=true

Ought to be solved by this PR: #1784

File: Makefile
Comment (damiannolan) [inline Makefile:412]: Is there release version of `confio/ics23` that we can use here?
I think it would best to avoid using a commit hash

Comment (fedekunze) [reply Makefile:412]: cc @ethanfrey

Comment (colin-axner) [reply Makefile:412]: Looks like 0.7.1 is the latest stable tag https://github.com/confio/ics23/tags

**Comment by colin-axner** [comment]
Hi @faddat, I made some suggested changes in this [pr](https://github.com/notional-labs/ibc-go/pull/75), feel free to merge if you agree with all the changes

**Comment by catShaark** [comment]
> Hi @faddat, I made some suggested changes in this [pr](https://github.com/notional-labs/ibc-go/pull/75), feel free to merge if you agree with all the changes

I'm looking at it now, thanks

File: modules/apps/transfer/keeper/relay_test.go
Comment (catShaark) [inline modules/apps/transfer/keeper/relay_test.go:155]: why are we changing this to math.Int

Comment (catShaark) [reply modules/apps/transfer/keeper/relay_test.go:155]: ohh, I see

File: Makefile
Comment (faddat) [reply Makefile:412]: sadly, it didn't work -- the correct release though, is v0.7.1

(no clue why it wouldn't work)

**Comment by faddat** [comment]
Colin, definitely agree with the changes!

One thing I'd like to contribute post-merge is a bit of additional refinement to the Makefile, that'll make sure that we can always get the imports back "the right way" should they stray.

**Comment by crodriguezvega** [comment]
@faddat @catShaark I have also opened a [PR](https://github.com/notional-labs/ibc-go/pull/76) with some changes that I think are needed to refactor the acknowledgement construction in the host chain of interchain accounts. Please also take a look!

**Review by colin-axner — APPROVED** [review]
Most everything looks great to me. Please apply my suggested changes to [ics27](https://github.com/notional-labs/ibc-go/pull/77)

`make proto-all` doesn't work on my machine. Is this true for anyone else? I see our script is outdated in comparison to the SDK (protocgen.sh was copied from the SDK) and they they are now using buf to maintain the proto files. I believe this is the same suggestion as #1345 

Going to give an approval conditioned on inclusion of my suggested changes to ics27 and that we fix proto generation in a followup pr (if it is broken). I'd be in favor of adding BSR support for v5 if it is quick and easy to do (thus allowing us to use a similar script to the SDK)

File: testing/simapp/app.go
Comment (catShaark) [reply testing/simapp/app.go:349]: I've added it back

**Comment by crodriguezvega** [comment]
I opened [this PR](https://github.com/notional-labs/ibc-go/pull/79), which, I think, fixes `make proto-all`. Please review.

**Comment by faddat** [comment]
I'll test it now, on a machine where it didn't work before.

**Comment by faddat** [comment]
Hey, I've merged your PR, and I'm now going to fix another issue in the Makefile.

**Comment by faddat** [comment]
All set -- so, running:

make proto-update-deps
make proto-all

in sequence now works for me perfectly. 

The change was to make sure that proofs.proto ended up in the root, how buf expected it. 

@vuong177 @nghuyenthevinh2000 @catShaark any additional thoughts on this one?

**Comment by crodriguezvega** [comment]
@faddat When I run `make proto-update-deps` on my machine I get this error:

```
sed: 1: "third_party/proto/proof ...": undefined label 'hird_party/proto/proofs.proto'
make: *** [proto-update-deps] Error 1
```

Does anybody experience the same error? 

Also, do you mind to run `make proto-swagger-gen` and push the changes in the `docs/client/swagger-ui/swagger.yaml`. There are some changes caused by the update of the cosmos proto files. Thanks!

**Comment by faddat** [comment]
happy to

that looks like I made a typo maybe?

**Comment by faddat** [comment]
Just tried update-deps

works fine on my mac and in gitpod

**Comment by faddat** [comment]
sed: 1: "third_party/proto/proof ...": undefined label 'hird_party/proto/proofs.proto'
make: *** [proto-update-deps] Error 1

if you got that somewhere... its gotta be somewhere right?

I am going to search the code...

**Comment by faddat** [comment]
Carlos, feels mildly silly to ask these q's but .... 

What os/setup do you use?
were you on the right branch?

stumped on makefile stuff

**Comment by crodriguezvega** [comment]
> Carlos, feels mildly silly to ask these q's but ....
> 
> What os/setup do you use? were you on the right branch?
> 
> stumped on makefile stuff

Hi @faddat. Thanks for checking!

I am on Big Sur and I was on the right branch. I talked to @damiannolan and he also had the same problem on his machine, but I have tried now to run `make proto-update-deps` on main and I get the same error, so I think i has nothing to do with your branch; it's probably something to do with my machine/setup.

**Comment by colin-axner** [comment]
The issue @crodriguezvega is facing is a mac issue with `sed`. Specifically this [line](https://github.com/cosmos/ibc-go/blob/6b4aa9a50096d69f1e392e25809f0efc15b3a5f4/Makefile#L465). We should help get this [issue](https://github.com/confio/ics23/issues/32) closed so that we can remove dependency on `sed`

`make proto-update-deps` works on my machine :+1: Also using `v0.7.1` works, but results in no diffs right now

File: testing/simapp/app.go
Comment (colin-axner) [reply testing/simapp/app.go:311]: Looks like we could use `sdk.GetConfig().GetBech32AccountAddrPrefix()` to appease the warning. I don't have a strong preference

**Review by colin-axner — APPROVED** [review]
Followup ACK. Excellent work @faddat and everyone else who contributed! Thanks for the speedy response time and applying all our suggestions

Looking forward to seeing this in a final release :rocket: :tada:

**Comment by faddat** [comment]
If it appeases it, let's do it.

**Review by damiannolan — APPROVED** [review]
Awesome work! And many thanks!

It would be nice to update the `CONFIO_URL` to point to the `v0.7.1` release version. 
Lets not block on the mac os `sed` issue, we can address it later or more importantly address the root cause from https://github.com/confio/ics23/issues/32

LGTM! 🚀

**Review by crodriguezvega — APPROVED** [review]
Awesome, @faddat! Thanks so much for this work; this has been a fun ride! 🤠

File: .gitpod.yml
Comment (colin-axner) [reply .gitpod.yml:1]: I think it got added back :sweat_smile:

**Pull request merged**
